### PR TITLE
fix: only prevent audio group creation if no other playlists are using it

### PR DESCRIPTION
**Pull request by brandonocasey**
Fixes an issue (not the only issue) with #977

caused by the fix in https://github.com/videojs/http-streaming/pull/952

### CODE REVIEW

**Review by gkatsev — APPROVED** [review]

File: src/media-groups.js [hunk 1]
```diff
@@ -405,18 +405,18 @@ export const initialize = {
...
         });
 
-        if (matchingPlaylists.length) {
-          // If there is a playlist that has the same uri as this audio variant, assume
-          // that the playlist is audio only. We delete the resolvedUri property here
-          // to prevent a playlist loader from being created so that we don't have
-          // both the main and audio segment loaders loading the same audio segments
-          // from the same playlist.
+        // If there are no playlists using this audio group other than ones
+        // that match it's uri. Then playlist is audio only. We delete the resolvedUri
```
Comment (gesinger) [inline src/media-groups.js:415]:
```suggestion
        // that match it's uri, then the playlist is audio only. We delete the resolvedUri
```

Comment (gesinger) [inline src/media-groups.js:415]: One thing I'm wondering is: what if we have a manifest where there are two playlists using the audio group, one is audio only, and another uses it as its demuxed audio for video. Wouldn't we need the audio loader running for that case?

**Review by gesinger — COMMENTED** [review]
Would probably be good to have some tests for this one, particularly since it might help document it.

File: src/media-groups.js (hunk 1 above)
Comment (brandonocasey) [reply src/media-groups.js:415]: We would but we don't support going from video/audio to just audio or vice versa, if I understand what you are saying.

**Comment by evanfarina** [comment]
Thanks for such a fast turnaround @brandonocasey , @gesinger and @gkatsev. I'll give this a shot first thing Monday.

**Comment by gkatsev** [comment]
This only fixes a portion of it. There's still the PTS weirdness.
Interestingly, with this fix, your stream doesn't start but it does work if I seek.

**Pull request merged**